datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StorageValidationService.java

```diff
@@ -68,6 +99,46 @@ public ObjectStorageValidateResponse validateObjectStorage(String credentialCrn,
                 () -> cloudProviderServicesV4Endpoint.validateObjectStorage(objectStorageValidateRequest));
     }
 
+    public ValidationResult validateBackupStorage(SdxCluster sdxCluster) {
+        DetailedEnvironmentResponse environmentResponse = environmentService.getDetailedEnvironmentResponseByName(sdxCluster.getEnvName());
+        CloudStorageRequest cloudStorageRequest = null;
+        try {
+            if (!StringUtils.isBlank(sdxCluster.getStackRequestToCloudbreak())) {
+                StackV4Request stackV4Request = JsonUtil.readValue(sdxCluster.getStackRequestToCloudbreak(), StackV4Request.class);
+                cloudStorageRequest = stackV4Request.getCluster().getCloudStorage();
+                LOGGER.info("Validating backup Storage: {}", JsonUtil.writeValueAsStringSilent(cloudStorageRequest));
+            }
+        } catch (IOException ioException) {
+            throw new BadRequestException("Failed to validate backup storage");
+        }
+
+        if (environmentResponse == null || cloudStorageRequest == null) {
+            throw new BadRequestException("Failed to validate backup storage");
+        }
+        ValidationResult.ValidationResultBuilder validationResultBuilder = new ValidationResult.ValidationResultBuilder();
+        try {
+            cloudStorageValidator.validateBackupLocation(cloudStorageRequest, environmentResponse, validationResultBuilder);
+        } catch (Exception e) {
+            String message = String.format("Error occured during object storage validation, validation skipped. Error: %s", e.getMessage());
+            LOGGER.warn(message);
+            sdxNotificationService.send(ResourceEvent.SDX_VALIDATION_FAILED_AND_SKIPPED, Set.of(e.getMessage()), sdxCluster);
+        }
+        ValidationResult validationResult = validationResultBuilder.build();
+        if (validationResult.hasError()) {
+            LOGGER.error(validationResult.getFormattedErrors());
+            eventSenderService.sendEventAndNotification(sdxCluster, sdxCluster.getInitiatorUserCrn(),
+                    ResourceEvent.SDX_VALIDATION_FAILED, Set.of(validationResult.getFormattedWarnings()));
+        }
+        if (validationResult.hasWarning()) {
+            LOGGER.info(validationResult.getFormattedWarnings());
+            eventSenderService.sendEventAndNotification(sdxCluster, sdxCluster.getInitiatorUserCrn(),
+                    ResourceEvent.SDX_VALIDATION_FAILED_AND_SKIPPED, Set.of(validationResult.getFormattedWarnings()));
```

Comment:
I was about to complain about `sdxCluster.getInitiatorUserCrn()`, because the creator user should usually not be employed for anything (the user might be already absent). On the other hand, as I see, all the methods of [EventSenderService](https://github.com/hortonworks/cloudbreak/blob/master/datalake/src/main/java/com/sequenceiq/datalake/events/EventSenderService.java) simply ignore the `userCrn` passed in the arg and instead obtain the user from `ThreadBasedUserCrnProvider.getUserCrn()` when constructing the structured event. Hence, I believe it would be beneficial to clean up `EventSenderService` to avoid further confusions. It is fine to have this tracked separately as there might be many references that will need changing (and possibly also require some elaboration among Raven folks).

Reply:
agree, initiatorUserCrn is deprecated, and there is a reason for that

Reply:
generally you should not use any deprecated getter or field at all

Reply:
regarding EventSenderService, I'll open a PR soon to address it

Reply:
https://github.com/hortonworks/cloudbreak/pull/13153/files

Reply:
I will not make any changes to this assuming that it be taken care of by https://github.com/hortonworks/cloudbreak/pull/13153/files.

Reply:
at least you should avoid usage of sdxCluster.getInitiatorUserCrn since as I mentioned it is deprecated and further usage should be avoided

Reply:
you can use null or empty string

Reply:
I have a PR which removes initiatorUserCrn from SdxCluster completely,so pls do not use it,it will cause conflicts in my PR

Reply:
https://github.com/hortonworks/cloudbreak/pull/12965

Reply:
Yes, please pass `null` or empty string here for the `userCrn`.

Reply:
ok. Will do

Reply:
addressed.